cpg-language-java/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/java/DeclarationHandler.kt

```diff
@@ -235,61 +235,64 @@ open class DeclarationHandler(lang: JavaLanguageFrontend) :
-                type = frontend.typeOf(variable.type)
-            } else {
-                type = this.objectType(t)
-                type.typeOrigin = Type.Origin.GUESSED
-            }
-        } catch (e: IllegalArgumentException) {
-            val t = frontend.recoverTypeFromUnsolvedException(e)
-            if (t == null) {
-                log.warn("Could not resolve type for {}", variable)
-                type = frontend.typeOf(variable.type)
-            } else {
-                type = this.objectType(t)
-                type.typeOrigin = Type.Origin.GUESSED
+
+        for (variable in fieldDecl.variables) {
+            val initializer =
+                variable.initializer
+                    .map { ctx: Expression -> frontend.expressionHandler.handle(ctx) }
+                    .orElse(null)
+                    as? de.fraunhofer.aisec.cpg.graph.statements.expressions.Expression
+            var type: Type
+            try {
+                // Resolve type first with ParameterizedType
+                type =
+                    frontend.typeManager.getTypeParameter(
+                        frontend.scopeManager.currentRecord,
+                        variable.resolve().type.describe()
+                    ) ?: frontend.typeOf(variable.resolve().type)
+            } catch (e: UnsolvedSymbolException) {
+                val t = frontend.recoverTypeFromUnsolvedException(e)
+                if (t == null) {
+                    log.warn("Could not resolve type for {}", variable)
+                    type = frontend.typeOf(variable.type)
+                } else {
+                    type = this.objectType(t)
+                    type.typeOrigin = Type.Origin.GUESSED
+                }
+            } catch (e: UnsupportedOperationException) {
+                val t = frontend.recoverTypeFromUnsolvedException(e)
+                if (t == null) {
+                    log.warn("Could not resolve type for {}", variable)
+                    type = frontend.typeOf(variable.type)
+                } else {
+                    type = this.objectType(t)
+                    type.typeOrigin = Type.Origin.GUESSED
+                }
+            } catch (e: IllegalArgumentException) {
+                val t = frontend.recoverTypeFromUnsolvedException(e)
+                if (t == null) {
+                    log.warn("Could not resolve type for {}", variable)
+                    type = frontend.typeOf(variable.type)
+                } else {
+                    type = this.objectType(t)
+                    type.typeOrigin = Type.Origin.GUESSED
+                }
             }
+            fieldDeclaration =
+                this.newFieldDeclaration(
+                    variable.name.asString(),
+                    type,
+                    modifiers,
+                    initializer,
+                    rawNode = fieldDecl
+                )
+            frontend.scopeManager.addDeclaration(fieldDeclaration)
```

Comment:
You can use the magical `DeclarationSequence` instead of the single field declaration. Not sure if we need to handle it at the caller. But that is how we do it in C++ if it can return more than one declaration